bug in Heliostat::getPanelById

As far as I can see, this function has never worked properly. g++ warns about this problem with the following messages:

../solarpilot/Heliostat.cpp:748:18: warning: left operand of comma operator has no effect [-Wunused-value]
../solarpilot/Heliostat.cpp:749:23: warning: left operand of comma operator has no effect [-Wunused-value]
../solarpilot/Heliostat.cpp:756:20: warning: left operand of comma operator has no effect [-Wunused-value]
../solarpilot/Heliostat.cpp:764:23: warning: left operand of comma operator has no effect [-Wunused-value]

I think this is a genuine bug, so submitting this issue (and I'll also submit a PR that fixes it).

--- a/solarpilot/Heliostat.cpp
+++ b/solarpilot/Heliostat.cpp
@@ -745,15 +745,14 @@ Reflector *Heliostat::getPanelById(int id){
 
 	for (int j=0; j<(int)ncantx; j++) {
 		for (int i=0; i<(int)ncanty; i++) {
-			if (_panels[j,i].getId() == id){
-				return &_panels[j,i];
+		  	if (_panels.at(j,i).getId() == id) {
+			  	return &_panels.at(j,i);
 			}
 		}
 	}
 
 	//#####call an error here
-
-	return &_panels[0,0];
+	return &_panels.at(0, 0);
 }
 
 Reflector *Heliostat::getPanel(int row, int col){
@@ -761,7 +760,7 @@ Reflector *Heliostat::getPanel(int row, int col){
 	nr = _panels.nrows();
 	nc = _panels.ncols();
 	if(row < nr && col < nc) {
-		return &_panels[row,col];
+	  	return &_panels.at(row, col);
 	}
 	else{
 		//FLAG -- this should be an error

